portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -31,15 +34,27 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   
-  private List<String> comments;
+//   private List<String> comments = new ArrayList<>();
 
-  @Override
-  public void init() {
-    comments = new ArrayList<>();
-  }
+//   @Override
+//   public void init() {
+//     comments = new ArrayList<>();
+//     }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> comments = new ArrayList<>();
+    for (Entity commentEntity : results.asIterable()) {
+      String txt = (String) commentEntity.getProperty("text-input");
+      long timestamp = (long) commentEntity.getProperty("timestamp");
```

Comment:
Does this code work?  I didn't think java had c-style casts..  getProperty returns a string, so you only need to cast the second one.  This would usually be done with Long.parseLong

Reply:
When I got rid of the casting I got incompatible types errors.  If I use the Long.parseLong methis, I would still have to cast the argument to a string.